ecs-cli/modules/config/config_test.go

```diff
@@ -14,86 +14,106 @@
 package config
 
 import (
+	"os"
 	"testing"
-
-	"github.com/aws/aws-sdk-go/aws/credentials"
 )
 
 const (
-	clusterName  = "defaultCluster"
-	profileName  = "defaultProfile"
-	region       = "us-west-1"
-	awsAccessKey = "AKID"
-	awsSecretKey = "SKID"
+	clusterName             = "defaultCluster"
+	profileName             = "defaultProfile"
+	region                  = "us-west-1"
+	awsAccessKey            = "AKID"
+	awsSecretKey            = "SKID"
+	envAwsAccessKey         = "envAKID"
+	envAwsSecretKey         = "envSKID"
+	credentialProviderCount = 5
 )
 
-func TestGetCredentialProvidersWithEmptyProfileAndCredentials(t *testing.T) {
+func TestGetCredentialProvidersVerifyProviderCountHasNotChanged(t *testing.T) {
 	ecsConfig := NewCliConfig(clusterName)
+	ecsConfig.Region = region
 	credentialProviders := ecsConfig.getCredentialProviders()
-	if len(credentialProviders) != 4 {
-		t.Error("Unexpected number of credential providers in the chain: ", len(credentialProviders))
+	if len(credentialProviders) != credentialProviderCount {
+		t.Fatal("Unexpected number of credential providers in the chain: ", len(credentialProviders))
 	}
 
-	// credentialProviders is composed of:
-	// 0 -> env provider
-	// 1 -> static creds provider
-	// 2 -> default profile provider
-	// 3 -> role provider
-	profileDefaultCredsProvider, ok := credentialProviders[2].(*credentials.SharedCredentialsProvider)
-	if !ok {
-		t.Fatal("Mismatch in credential provider chain. Expected to use 'default' profile creds provider")
-	}
-	// This would be the 'default' profile creds provider in this case.
-	if "" != profileDefaultCredsProvider.Profile {
-		t.Errorf("Invalid profile name set. Expected empty string. Got [%s]", profileDefaultCredsProvider.Profile)
-	}
 }
 
-func TestGetCredentialProvidersWithProfile(t *testing.T) {
+func TestToServiceConfigWhenUsingEnvVariables(t *testing.T) {
 	ecsConfig := NewCliConfig(clusterName)
-	ecsConfig.AwsProfile = profileName
-	credentialProviders := ecsConfig.getCredentialProviders()
-	if len(credentialProviders) != 3 {
-		t.Fatal("Unexpected number of credential providers in the chain: ", len(credentialProviders))
+	ecsConfig.Region = region
+
+	os.Setenv("AWS_ACCESS_KEY_ID", envAwsAccessKey)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", envAwsSecretKey)
+
+	// Clear env variables as they persist past the individual test boundary
+	defer func() {
+		os.Unsetenv("AWS_ACCESS_KEY_ID")
```

Comment:
If our tests are free of other environment dependencies, `os.Clearenv()` will be more readable.